src/tsx/components/inventory.tsx

```diff
@@ -0,0 +1,210 @@
+import * as React from "react";
+import { IComponentState, IEmpty } from "../types";
+
+import style from "../../less/components/inventory.module.less";
+import { OnTheFlyAction } from "../coreMessages";
+
+interface IProps {
+  inventory: IComponentState;
+  onOnTheFlyAction: (action: OnTheFlyAction) => void;
+}
+
+interface IItem {
+  name: string;
+  description: string;
+  cost: number;
+}
+
+interface IAchievement {
+  title: string;
+  desc: string;
+  name: string;
+  reuse: boolean;
+}
+
+interface IConfiguration {
+  achievements: IAchievement[];
+  items: IItem[];
+}
+
+class InventoryView extends React.PureComponent<IProps, IEmpty> {
+  public render(): JSX.Element {
+    const {
+      configuration,
+      money,
+      itemCount,
+      achievementsReached,
+      achievementNames,
+    } = this._prepareData();
+
+    return (
+      <div className={style.container}>
+        {achievementsReached
```

Comment:
How much will it help having the reached achievements up here? I'm thinking we will spend most of the time scrolled down somewhere, and then this just means everything shifts a bit.

Reply:
I think it doesn't affect the scrolling so much. You will never scroll past it (since if it's there it should be activated) and the view doesn't move scroll-wise when it is added. So overall I don't think it has a negative effect, and a slight positive effect.

Reply:
Fair enough, approved!